### ISSUE TITLE
[Bug]: Currency Input widget always shows error

### ISSUE THREAD

**Issue by aswathkk** [appsmithorg/appsmith#14862]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The currency widget shows "Invalid Input" even when we do not set any validation logic

https://user-images.githubusercontent.com/10436935/176142869-237ba43b-f9ec-4d98-8833-4b86e6243d19.mov

This happens only in the attached application

[Currency Input Issue.json.zip](https://github.com/appsmithorg/appsmith/files/8999584/Currency.Input.Issue.json.zip)

### Steps To Reproduce

Add a currency input widget and try to add some values to see the error

### Public Sample App

https://app.appsmith.com/app/currency-input-issue/page1-62bac69f20ae3225cef2fb2f

### Version

1.7.5

**Comment by sbalaji1192** [comment]
@techbhavin @dilippitchika @aswathkk This issue is happening when the evaulationVersion of the application is 1.

**Comment by dilippitchika** [comment]
@sbalaji1192 what does it mean in layman terms, is this possible for someone to face this in production?

**Comment by sbalaji1192** [comment]
@dilippitchika Yes. If they start using the currency widget in an old application. it should happen in production as well. Today I will look into a possible solution for this issue.

**Comment by Nikhil-Nandagopal** [comment]
@sbalaji1192 I suggest we don't support older apps for this widget then and simply ask users to upgrade

**Comment by sbalaji1192** [comment]
@techbhavin We can tweak the regex to make it work in both evaluated versions. let me know if this works.
```
text
  .replace(new RegExp(`[${getLocaleThousandSeparator()}]`, "g"), "")
  .replace(new RegExp(`[${getLocaleDecimalSeperator()}]`), "."),
```
